newrelic-agent/src/main/java/com/newrelic/agent/HarvestServiceImpl.java

```diff
@@ -96,25 +96,15 @@ public void startHarvestables(IRPMService rpmService, AgentConfig config) {
                 int maxSamplesStored = tracker.harvestable.getMaxSamplesStored();
                 long reportPeriodInMillis = HarvestServiceImpl.REPORTING_PERIOD_IN_MILLISECONDS;
                 boolean isSpanEventEndpoint = tracker.harvestable.getEndpointMethodName().equals(SPAN_EVENT_DATA);
-                boolean isLogEventEndpoint = tracker.harvestable.getEndpointMethodName().equals(LOG_EVENT_DATA);
 
-                // The event_harvest_config.harvest_limits received from server-side during the connect lifecycle contains config for error_event_data,
-                // analytic_event_data, and custom_event_data TODO The event_harvest_config.harvest_limits should also include log_event_data endpoint at some point...
+                // The event_harvest_config.harvest_limits received from server-side during the connect lifecycle
+                // contains config for error_event_data, analytic_event_data, custom_event_data, and log_event_data
                 if (eventHarvestConfig != null && !isSpanEventEndpoint) {
                      Agent.LOG.log(Level.FINE, "event_harvest_config from collector is: {0} samples stored for {1}", maxSamplesStored,
                             tracker.harvestable.getEndpointMethodName());
                     Map<String, Object> harvestLimits = (Map<String, Object>) eventHarvestConfig.get(HARVEST_LIMITS);
 
-                    Long harvestLimit;
-                    // FIXME THIS IS A HACK! Real harvestLimit for log_event_data endpoint should come from server side event_harvest_config.harvest_limits!!!
-                    if (isLogEventEndpoint) {
-                        int logEventHarvestPeriodInSeconds = 5;
-                        int harvestsPerMinute = 60 / logEventHarvestPeriodInSeconds;
-                        int maxSamplesStoredLogEvents = ServiceFactory.getLogSenderService().getMaxSamplesStored() / harvestsPerMinute;
-                        harvestLimit = (long) maxSamplesStoredLogEvents;
-                    } else {
-                        harvestLimit = (Long) harvestLimits.get(tracker.harvestable.getEndpointMethodName());
-                    }
+                    Long harvestLimit = (Long) harvestLimits.get(tracker.harvestable.getEndpointMethodName());
```

Comment:
For my edification...Is there a particular reason you chose to cast to Long for this one?  I know something changed after Java 1.7, but can't find exactly what. But, also noticed the older code in this class casts to a primitive long.  Is it just a personal choice since autoboxing was introduced or is there more to it?

Reply:
This particular line has been cast to a `Long` for at least three years so I'm just maintaining that behavior. We could probably refactor to use a primitive `long` but then it invalidates the `if (harvestLimit != null)` check.

https://github.com/newrelic/newrelic-java-agent/blob/3b934f3ccfbd15437a2f85ef705979148ab5a293/newrelic-agent/src/main/java/com/newrelic/agent/HarvestServiceImpl.java#L104